src/main/java/com/atlassian/db/replica/api/KeyBasedThrottledCache.java

```diff
@@ -0,0 +1,27 @@
+package com.atlassian.db.replica.api;
+
+import com.atlassian.db.replica.internal.util.ThreadSafe;
+import com.atlassian.db.replica.spi.SuppliedCache;
+
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
+
+/**
+ * Allows for thread-safe key-based access to ThrottledCache.
+ * It can be used to have only one instance per key (e.g. database url).
+ */
+@ThreadSafe
+public class KeyBasedThrottledCache implements SuppliedCache<Long> {
+    private static final ConcurrentHashMap<String, ThrottledCache<Long>> caches = new ConcurrentHashMap<>();
```

Comment:
```suggestion
    private static final ConcurrentHashMap<String, SuppliedCache<T>> caches = new ConcurrentHashMap<>();
```
Now it can work with any `SuppliedCache`, so we could inject `Supplier<SuppliedCache<T>>`, freeing the consumer to use different strategies while preserving the keying. And also holding different values than `Long`.